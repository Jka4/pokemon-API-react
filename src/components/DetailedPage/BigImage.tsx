import React, { lazy } from 'react';
import Paper from '@material-ui/core/Paper';

import { Pokes } from 'types';

const ImageContainer = lazy(() => import('components/ImageContainer/ImageContainer'));

interface RootType {
  bigImage?: Pokes;
}

const BigImage: React.FC<RootType> = ({ bigImage }: RootType) => {
  const placeholder = () => {
    return <img src={bigImage?.placeholderBase64} className="bigImage " alt="" style={{ filter: 'blur(18px)' }} />;
  };

  return (
    <Paper elevation={3} style={{ overflow: 'hidden' }}>
      <ImageContainer url={bigImage?.imageHQ || ''} cn={'bigImage'} fallback={placeholder} />
    </Paper>
  );
};

export default BigImage;
