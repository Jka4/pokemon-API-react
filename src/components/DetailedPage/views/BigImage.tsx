import React from 'react';

import { PokesTypes } from 'types/index';
import styled from 'styled-components';
import ImageContainer from 'components/ImageContainer/ImageContainer';

interface PropsType {
  bigImage?: PokesTypes;
}

const BigImage: React.FC<PropsType> = ({ bigImage }: PropsType) => {
  const biImageDefaultValue =
    'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABg0lEQVQ4T2NkIBMwYtEHE/uPz0x0jSA+WMP+/Q0sDAwHGBwdHf4xMDT8QzcEWSNUk44sA8OVvwwMDM9giv//D2VmZFwNEoMDuMb9DQwsjg0Mf/qzmWMVpC0tAyoTarYtW5LnHf37PgPD0YUgHf8ZGBgZoS6Ca1y1KpQ5LGz131WZDMbmiQmnOXlZGK9fvbH96UfuuW4WTBFndj5K9yi6+u7/fwZGRkaG/0g2OrA4Nhz4s67S1N7K8esBQSYGhgu77s3RiTc3/cLEo71v20OlyOIrj///b2BiZGz4B9f4/z8DEyMjw7/OTj2ZeCvuKyL8jPwP3jPfeXfpwrtDFz6HlcxleAhTA3I2wsb9DiyOjgf+HFtmEGBiLbv+PzMTw/enrxhmNR2fWbaVIQNZE1aNeycqJErJcFfycDNc+/rmg75GzFNzBgaGVw0NDEwNDQzwaMGSAER5GBhe/2RgYPhtbSwld/Tss0fYEgLOBAALBOQowBqP8MiGxBU42tCdh1cjsWkeAIUZlA82LlHPAAAAAElFTkSuQmCC';

  const placeholder = () => {
    return <img src={bigImage?.placeholderBase64} alt="" style={{ filter: 'blur(18px)' }} />;
  };

  return (
    <ImageWrapper>
      <ImageContainer url={bigImage?.imageHQ || biImageDefaultValue} cn={'deBlur'} fallback={placeholder} />
    </ImageWrapper>
  );
};

const ImageWrapper = styled.div`
  background: #2196f3;
  border-radius: 5px;
  border: 1px solid black;
  width: 425px;
  height: 430px;
  overflow: hidden;
  box-shadow: 0 3px 3px -2px rgb(0 0 0 / 20%), 0px 3px 4px 0px rgb(0 0 0 / 14%), 0px 1px 8px 0px rgb(0 0 0 / 12%);

  img {
    width: 100%;
    height: auto;
  }

  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    width: 100%;
  }
`;

export default BigImage;
