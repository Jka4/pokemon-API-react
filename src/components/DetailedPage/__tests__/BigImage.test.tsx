import BigImage from '../views/BigImage';

import { mount } from 'enzyme';

describe('<BigImage/> ', () => {
  const fakeData = {
    id: 168,
    name: 'ariados',
    weight: 335,
    image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/168.png',
    imageHQ: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/168.png',
    chain: [
      {
        species_name: 'spinarak',
      },
      {
        species_name: 'ariados',
      },
    ],
    placeholderBase64:
      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABiklEQVQ4T2NkIBMw4tAHEwfR/7CpwaYRJPafkEPQNYI1+ZqZKX7n4/vL/OcP284DBx4wMDD8QTcIRWNa2kzWWbPSfzPwiqQyfH6jwsDAcJaBgWGVg4MDy4EDB1A0wzWuWrWKOSws7G/07G0ydZeqtjKwy+kdl3bJSijMmw61DcULyDaCJVbMmjUpiP9m7tOfAv9OXfrI9OnMugmpB+4V/mdgYGRE8jtYI8y246ePu9w+c73LSOe5AjMTs8DRzIrfSqmT2H5oaJZ6ubr27N+/n8XR0RHsZLDG////MzIyMv6/fWIJ39zN7/oNhG+6B3grSb96/5vh/P5vDDzapnOc/XxT/69axcwYFvYXQ+OqVTP5v31izXr9/Im7s+FvEaa/X/9ceaygL6qqWebu7toNcxlcI7KtIHZnZ69FeXnxRQYGhu+LFyyL/PHp8eHUvPInDQ0NTA0NDeAEgRId//8zMB5o3M/s2ADxBzKAeQcmhjXJQZ0ENnnVqlVMoaGh/0BhgGwQrrRKKMUxAADfoqQPKXvh9QAAAABJRU5ErkJggg==',
    order: 250,
    base_experience: 140,
  };

  // it('renders with data', () => {
  //   const container = mount(<BigImage poke={fakeData} />);
  //   expect(container).toMatchSnapshot();
  // });

  it('renders without data', () => {
    const container = mount(<BigImage imageHQ={fakeData.imageHQ} placeholderBase64={fakeData.placeholderBase64} />);
    expect(container).toMatchSnapshot();
  });
});
