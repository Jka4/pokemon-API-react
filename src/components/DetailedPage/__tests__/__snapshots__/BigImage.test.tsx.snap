// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigImage/>  renders without data 1`] = `
<BigImage
  imageHQ="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/168.png"
  placeholderBase64="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABiklEQVQ4T2NkIBMw4tAHEwfR/7CpwaYRJPafkEPQNYI1+ZqZKX7n4/vL/OcP284DBx4wMDD8QTcIRWNa2kzWWbPSfzPwiqQyfH6jwsDAcJaBgWGVg4MDy4EDB1A0wzWuWrWKOSws7G/07G0ydZeqtjKwy+kdl3bJSijMmw61DcULyDaCJVbMmjUpiP9m7tOfAv9OXfrI9OnMugmpB+4V/mdgYGRE8jtYI8y246ePu9w+c73LSOe5AjMTs8DRzIrfSqmT2H5oaJZ6ubr27N+/n8XR0RHsZLDG////MzIyMv6/fWIJ39zN7/oNhG+6B3grSb96/5vh/P5vDDzapnOc/XxT/69axcwYFvYXQ+OqVTP5v31izXr9/Im7s+FvEaa/X/9ceaygL6qqWebu7toNcxlcI7KtIHZnZ69FeXnxRQYGhu+LFyyL/PHp8eHUvPInDQ0NTA0NDeAEgRId//8zMB5o3M/s2ADxBzKAeQcmhjXJQZ0ENnnVqlVMoaGh/0BhgGwQrrRKKMUxAADfoqQPKXvh9QAAAABJRU5ErkJggg=="
>
  <styled.div>
    <div
      className="sc-eCssSg gJZQPD"
    >
      <ImageContainer
        fallback={
          <img
            alt=""
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABiklEQVQ4T2NkIBMw4tAHEwfR/7CpwaYRJPafkEPQNYI1+ZqZKX7n4/vL/OcP284DBx4wMDD8QTcIRWNa2kzWWbPSfzPwiqQyfH6jwsDAcJaBgWGVg4MDy4EDB1A0wzWuWrWKOSws7G/07G0ydZeqtjKwy+kdl3bJSijMmw61DcULyDaCJVbMmjUpiP9m7tOfAv9OXfrI9OnMugmpB+4V/mdgYGRE8jtYI8y246ePu9w+c73LSOe5AjMTs8DRzIrfSqmT2H5oaJZ6ubr27N+/n8XR0RHsZLDG////MzIyMv6/fWIJ39zN7/oNhG+6B3grSb96/5vh/P5vDDzapnOc/XxT/69axcwYFvYXQ+OqVTP5v31izXr9/Im7s+FvEaa/X/9ceaygL6qqWebu7toNcxlcI7KtIHZnZ69FeXnxRQYGhu+LFyyL/PHp8eHUvPInDQ0NTA0NDeAEgRId//8zMB5o3M/s2ADxBzKAeQcmhjXJQZ0ENnnVqlVMoaGh/0BhgGwQrrRKKMUxAADfoqQPKXvh9QAAAABJRU5ErkJggg=="
            style={
              Object {
                "filter": "blur(18px)",
              }
            }
          />
        }
        url="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/168.png"
      >
        <ErrorBoundary>
          <styled.img
            loading="lazy"
            ready={false}
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/168.png"
          >
            <img
              className="sc-hKgILt hmiyim"
              loading="lazy"
              src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/168.png"
            />
          </styled.img>
          <styled.div
            ready={false}
          >
            <div
              className="sc-dlfnbm fYocxX"
            >
              <img
                alt=""
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABiklEQVQ4T2NkIBMw4tAHEwfR/7CpwaYRJPafkEPQNYI1+ZqZKX7n4/vL/OcP284DBx4wMDD8QTcIRWNa2kzWWbPSfzPwiqQyfH6jwsDAcJaBgWGVg4MDy4EDB1A0wzWuWrWKOSws7G/07G0ydZeqtjKwy+kdl3bJSijMmw61DcULyDaCJVbMmjUpiP9m7tOfAv9OXfrI9OnMugmpB+4V/mdgYGRE8jtYI8y246ePu9w+c73LSOe5AjMTs8DRzIrfSqmT2H5oaJZ6ubr27N+/n8XR0RHsZLDG////MzIyMv6/fWIJ39zN7/oNhG+6B3grSb96/5vh/P5vDDzapnOc/XxT/69axcwYFvYXQ+OqVTP5v31izXr9/Im7s+FvEaa/X/9ceaygL6qqWebu7toNcxlcI7KtIHZnZ69FeXnxRQYGhu+LFyyL/PHp8eHUvPInDQ0NTA0NDeAEgRId//8zMB5o3M/s2ADxBzKAeQcmhjXJQZ0ENnnVqlVMoaGh/0BhgGwQrrRKKMUxAADfoqQPKXvh9QAAAABJRU5ErkJggg=="
                style={
                  Object {
                    "filter": "blur(18px)",
                  }
                }
              />
            </div>
          </styled.div>
        </ErrorBoundary>
      </ImageContainer>
    </div>
  </styled.div>
</BigImage>
`;
